src/ui/screen/screen_setup_mods.cpp

```diff
@@ -0,0 +1,281 @@
+
+    if (event.type == EVENT_INTERFACE_LOAD)
+    {
+        pl = static_cast<CList*>(pw->SearchControl(EVENT_INTERFACE_MODS_UNLOADED));
+        if ( pl == nullptr )  return false;
+        modName = pl->GetItemName(pl->GetSelect());
+
+        modPathRaw = CResourceManager::GetSaveLocation() + "/" + "mods" + "/";
+        modPath = modPathRaw.c_str();
+        boost::filesystem::rename(modPath+OFF+modName, modPath+modName);
+        m_pathManager->AddMod(modPath+modName);
+        m_app->Reload();
+        m_main->ChangePhase(PHASE_SETUPm);
+    }
+    if (event.type == EVENT_INTERFACE_UNLOAD)
+    {
+        pl = static_cast<CList*>(pw->SearchControl(EVENT_INTERFACE_MODS_LOADED));
+        if ( pl == nullptr )  return false;
+        modName = pl->GetItemName(pl->GetSelect());
+
+        modPathRaw = CResourceManager::GetSaveLocation() + "/" + "mods" + "/";
+        modPath = modPathRaw.c_str();
+        m_pathManager->RemoveMod(modPath+modName);
+        boost::filesystem::rename(modPath+modName, modPath+OFF+modName);
+        m_app->Reload();
+        m_main->ChangePhase(PHASE_SETUPm);
+    }
+    if (event.type == EVENT_INTERFACE_MODS_DIR)
+    {
+        modPathRaw = CResourceManager::GetSaveLocation() + "/" + "mods";
+        #if defined(PLATFORM_WINDOWS)
+            result = system(("start \""+modPathRaw+"\"").c_str());
+        #elif defined(PLATFORM_LINUX)
+            result = system(("xdg-open \""+modPathRaw+"\"").c_str());
+        #elif defined(PLATFORM_MACOSX)
+            result = system(("open \""+modPathRaw+"\"").c_str());
+        #endif
+        if (result == -1)
+        {
+            GetLogger()->Error("Failed to open Mods directory! Does directory exists?\n");
+        }
+    }
+    switch (event.type)
+    {
+        case EVENT_INTERFACE_MODS_UNLOADED:
+            pl = static_cast<CList*>(pw->SearchControl(EVENT_INTERFACE_MODS_LOADED));
+            if ( pl == nullptr )  break;
+
+            pb = static_cast<CButton*>(pw->SearchControl(EVENT_INTERFACE_UNLOAD));
+            if ( pb == nullptr )  break;
+            pl->SetSelect(-1);
+            pb->ClearState(STATE_ENABLE);
+
+            pb = static_cast<CButton*>(pw->SearchControl(EVENT_INTERFACE_LOAD));
+            if ( pb == nullptr )  break;
+            pb->SetState(STATE_ENABLE);
+            break;
+
+        case EVENT_INTERFACE_MODS_LOADED:
+            pl = static_cast<CList*>(pw->SearchControl(EVENT_INTERFACE_MODS_UNLOADED));
+            if ( pl == nullptr )  break;
+
+            pb = static_cast<CButton*>(pw->SearchControl(EVENT_INTERFACE_LOAD));
+            if ( pb == nullptr )  break;
+            pl->SetSelect(-1);
+            pb->ClearState(STATE_ENABLE);
+
+            pb = static_cast<CButton*>(pw->SearchControl(EVENT_INTERFACE_UNLOAD));
+            if ( pb == nullptr )  break;
+            pb->SetState(STATE_ENABLE);
+            break;
+
+        case EVENT_INTERFACE_WORKSHOP:
+            #if defined(PLATFORM_WINDOWS)
```

Comment:
Files other than system_*.cpp should not contain any platfrom-dependent #ifs, please move this to a function in the system module